src/backend/tools/base.py

```diff
@@ -31,7 +31,88 @@ class ToolError(BaseModel, extra="allow"):
     text: str
     details: str = ""
 
-class BaseTool():
+def check_type(param_value, type_description: str) -> bool:
```

Comment:
Nit: can we move check_type, check_type_recursive and check_tool_parameters into the tool.utils folder, then keep the metaclass here